src/main/java/gregtech/api/recipes/RecipeMaps.java

```diff
@@ -125,15 +129,43 @@ public class RecipeMaps {
      *               .fluidInputs(Materials.SolderingAlloy.getFluid(GTValues.L))
      *               .fluidInputs(Materials.Lubricant.getFluid(250))
      *               .output(MetaItems.ELECTRIC_MOTOR_LuV)
+     *               .research(ELECTRIC_MOTOR_IV.getStackForm())
      *               .duration(600).EUt(6000).buildAndRegister();
      * </pre>
      *
-     * The Assembly Line Recipe Builder has no special properties/build actions yet, but will in the future
+     * The Assembly Line has a special action that is performed when the recipe is built, designated by the <B>onRecipeBuild</B>
+     * call on the Recipe Map. This action adds a Scanner recipe for the specified research item, if generation of this recipe is enabled.
      */
     @ZenProperty
-    public static final RecipeMapAssemblyLine<SimpleRecipeBuilder> ASSEMBLY_LINE_RECIPES = (RecipeMapAssemblyLine<SimpleRecipeBuilder>) new RecipeMapAssemblyLine<>("assembly_line", 16, false, 1, false,  4, false,  0, false, new SimpleRecipeBuilder(), false)
+    public static final RecipeMap<AssemblyLineRecipeBuilder> ASSEMBLY_LINE_RECIPES = new RecipeMapAssemblyLine("assembly_line", 16, 1, 4, 0, new AssemblyLineRecipeBuilder(), false)
             .setProgressBar(GuiTextures.PROGRESS_BAR_ARROW, MoveType.HORIZONTAL)
-            .setSound(GTSoundEvents.ASSEMBLER);
+            .setSound(GTSoundEvents.ASSEMBLER)
+            .onRecipeBuild(recipeBuilder -> {
+                if (!ConfigHolder.machines.enableResearch) return;
+
+                AssemblyLineRecipeBuilder builder = (AssemblyLineRecipeBuilder) recipeBuilder;
+                if (!builder.shouldAddResearchRecipe()) return;
+
+                String researchId = builder.getResearchId();
+                if (researchId.isEmpty()) return;
+
+                NBTTagCompound compound = new NBTTagCompound();
+                compound.setTag(IResearchRecipeMap.RESEARCH_NBT_TAG, AssemblyLineRecipeBuilder.generateResearchNBT(researchId));
+
+                ItemStack dataStick = MetaItems.TOOL_DATA_STICK.getStackForm();
+                dataStick.setTagCompound(compound);
+
+                ItemStack researchItem = builder.getResearchItem();
+                if (researchItem.isEmpty()) return;
+
+                RecipeMaps.SCANNER_RECIPES.recipeBuilder()
+                        .input(MetaItems.TOOL_DATA_STICK)
```

Comment:
Should we allow for any NBT on the data stick, to override existing data stick recipes?